src/sinks/sematext.rs

```diff
@@ -0,0 +1,112 @@
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct SematextConfig {
+    cloud: Cloud,
+    #[serde(flatten)]
+    inner: ElasticSearchConfig,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Cloud {
+    NorthAmerica,
+    Europe,
+}
+
+#[typetag::serde(name = "sematext")]
+impl SinkConfig for SematextConfig {
+    fn build(&self, cx: SinkContext) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
+        let mut es_config = self.inner.clone();
+
+        let host = match &self.cloud {
+            Cloud::NorthAmerica => "https://logsene-receiver.sematext.com",
+            Cloud::Europe => "https://logsene-receiver.sematext.com",
+        };
+
+        // Custom config overrides
+        if es_config.host.is_none() {
+            es_config.host = Some(host.to_string());
+        }
+
+        if es_config.index.is_none() {
+            Err(format!("`index` field is required"))?;
+        }
+
+        if es_config.doc_type.is_none() {
+            es_config.doc_type = Some("logs".to_string());
+        }
+
+        if es_config.compression.is_none() {
+            es_config.compression = Some(Compression::None);
+        }
+
+        es_config.build(cx)
+    }
+
+    fn input_type(&self) -> DataType {
+        self.inner.input_type()
+    }
+
+    fn sink_type(&self) -> &'static str {
+        "sematext"
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::event::Event;
+    use crate::sinks::util::test::{build_test_server, load_sink};
+    use crate::test_util;
+    use crate::topology::config::SinkConfig;
+    use futures::{Sink, Stream};
+
+    #[test]
+    fn smoke() {
+        let (mut config, cx, mut rt) = load_sink::<SematextConfig>(
+            r#"
+           cloud = "north_america"
+            index = "mylogtoken"
+        "#,
+        )
+        .unwrap();
+
+        // Make sure we can build the config
+        let _ = config.build(cx.clone()).unwrap();
+
+        let addr = test_util::next_addr();
+        // Swap out the host so we can force send it
+        // to our local server
+        config.inner.host = Some(format!("http://{}", addr));
+
+        let (sink, _) = config.build(cx).unwrap();
+
+        let (rx, _trigger, server) = build_test_server(&addr);
+        rt.spawn(server);
+
+        let (expected, lines) = test_util::random_lines_with_stream(100, 10);
+        let pump = sink.send_all(lines.map(Event::from));
+        let _ = rt.block_on(pump).unwrap();
+
+        let output = rx.take(1).wait().collect::<Result<Vec<_>, _>>().unwrap();
+
+        let val = serde_json::Deserializer::from_slice(&output[0].1[..])
+            .into_iter::<serde_json::Value>()
+            .enumerate()
+            .filter(|(i, _)| i % 2 == 1)
+            .map(|v| v.1.unwrap())
+            .map(|v| v.get("message").unwrap().as_str().unwrap().to_string())
+            .enumerate()
+            .collect::<Vec<_>>();
```

Comment:
This whole thing is really confusing. It's unclear what type `output` is, what we're getting out of `&output[0].1[..]` (is this Perl? :smile:), why we're taking the odd numbered ones, and what is getting unwrapped.

Reply:
I want to make this easier :) I'm prob gonna write a few utils to make this a bit cleaner and easier to read. But this removes all even messages and then extracts their message field.

Reply:
Why are we only taking the even messages?

Reply:
ES sends things as ndjson where one line is index info and second line is the actual message object.

Reply:
Right, a comment about that would have been great :smile: 

This is kinda nitpicky, but having readable tests is really important. I'd have pulled out `&output[0].1[..]` at least to its own binding, giving it a name (e.g. `body`) if nothing else. Then commented why we only wanted the odd lines. Then tried to make it a little clearer that `into_iter::<Value>` is actually an iterator of `Result<Value, _>` and that's what we're unwrapping later.

Little things like that can really help readers cement their understanding of what's being tested and therefore how the component should be behaving.

Reply:
> There is also a test, I plan on cleaning it up a bit more because I think the giant iterator I have will be more common.

:)

Reply:
but yeah, I think there are some nice things we can do here this was just to get it working for now. I agree with your feedback though!

Reply:
Going to punt on this one for now as I want to explore what other similar sink iterators look like to see what makes sense to keep specifically locally within this test file.